src/sharing/components/EditableSharingModal.jsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react'
+import flow from 'lodash/flow'
+import { queryConnect, withClient } from 'cozy-client'
+
+import { Contact, Group } from 'cozy-doctypes'
+
+import SharingContext from 'sharing/context'
+import ContactsAndGroupsDataLoader from 'sharing/components/ContactsAndGroupsDataLoader'
+import { default as DumbShareModal } from 'sharing/components/ShareModal'
+
+const EditableSharingModal = ({
+  client,
+  contacts,
+  document,
+  groups,
+  ...rest
+}) => {
+  return (
+    <ContactsAndGroupsDataLoader contacts={contacts} groups={groups}>
+      <SharingContext.Consumer>
+        {({
+          documentType,
+          isOwner,
+          getRecipients,
+          getSharingLink,
+          share,
+          revoke,
+          shareByLink,
+          revokeSharingLink,
+          hasSharedParent,
+          hasSharedChild
+        }) => (
+          <DumbShareModal
+            document={document}
+            documentType={documentType}
+            contacts={contacts}
+            createContact={contact => client.create(Contact.doctype, contact)}
+            groups={groups}
+            recipients={getRecipients(document.id)}
+            link={getSharingLink(document.id)}
+            isOwner={isOwner(document.id)}
+            hasSharedParent={hasSharedParent(document)}
+            hasSharedChild={hasSharedChild(document)}
+            onShare={share}
+            onRevoke={revoke}
+            onShareByLink={shareByLink}
+            onRevokeLink={revokeSharingLink}
+            {...rest}
+          />
+        )}
+      </SharingContext.Consumer>
+    </ContactsAndGroupsDataLoader>
+  )
+}
+
+const contactsQuery = client =>
+  client
+    .all(Contact.doctype)
+    .where({
+      _id: {
+        $gt: null
+      },
+      // trashed: {
+      //   $exists: false
```

Comment:
Nit: if we don't need this, we can remove the comment :)

Reply:
we will in fact need it but we have another card for this so I'll remove the comment